ls1intum/Artemis
# `Exam mode`: Fix an issue when preparing the exercises shortly before the exam start

### Checklist
#### General
#### Server

### Motivation and Context
If an exam participation gets created shortly before the exam starts, the locking of this repository does not get scheduled. 

### Description
This PR reuses the new `sendRescheduleAllStudentExams` message to reschedule all locking tasks at the exam end date.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exam with programming exercises

1. Create an exam with a start date shortly in the future.
2. Edit the individual working time of different students
3. Press `Prepare Exercise start` with less than 5 minutes left until the start date.
4. Ensure that at the end of the exam, all repositories get locked. Ensure that the individual end date gets accounted for here.

### Review Progress

#### Code Review
#### Manual Tests
